octopuslite/reader.py

```diff
@@ -104,43 +103,54 @@ def __getitem__(self, channel_name: Union[str, Channels]):
 
         return self._lazy_arrays[channel_name]
 
-
     def files(self, channel_name: str) -> list:
         return self._files[Channels[channel_name.upper()]]
 
     def _load_and_process(self, fn: str) -> np.ndarray:
         """Load and crop the image."""
         image = io.imread(fn)
 
-        t = int(parse_filename(fn)['time'])
-        image = self._transformer(image, t)
+        if self._transformer is not None:
+            # need to use index of file as some frames may have been removed
+            channel = parse_filename(fn)["channel"]
+            files = [
+                os.path.join(self.path, f)
+                for f in os.listdir(self.path)
+                if f"channel{str(channel.value).zfill(3)}" in f
+                and f.endswith((".tif", ".tiff"))
+            ]
```

Comment:
Can you replace this using the inbuilt `files`?
```suggestion
            channel = parse_filename(fn)["channel"]
            files = self.files(channel)
```

Reply:
yep, oop didnt realise you did it there. i used `channel.name` instead

Reply:
needs to be `channel.name` anyway